src/modules/azure-azcopy-sync.ts

```diff
@@ -0,0 +1,134 @@
+import { Echo } from "./echo";
+import shell from "shelljs";
+import { CommandStringBuilder } from "../utilities/command-string-builder";
+import child_process from "child_process";
+
+// -----------------------------------------------------------------------------------------
+// #region Constants
+// -----------------------------------------------------------------------------------------
+
+const AZCOPY_COMMAND = "azcopy sync";
+
+// #endregion Constants
+
+// -----------------------------------------------------------------------------------------
+// #region Interfaces
+// -----------------------------------------------------------------------------------------
+
+interface BlobStorageUrlParts {
+    account: string;
```

Comment:
Consider adding some additional JSDoc commentary on what these fields represent - not super familiar with the Azure API, but is this an account name? Some sort of namespaced string? 🤔